crates/astria-conductor/src/execution_client.rs

```diff
@@ -1,88 +1,154 @@
-use color_eyre::eyre::Result;
+use color_eyre::eyre::{
+    Result,
+    WrapErr,
+};
 use prost_types::Timestamp;
-use proto::generated::execution::v1alpha1::{
+use proto::generated::execution::v1alpha2::{
     execution_service_client::ExecutionServiceClient,
-    DoBlockRequest,
-    DoBlockResponse,
-    FinalizeBlockRequest,
-    InitStateRequest,
-    InitStateResponse,
+    BatchGetBlocksRequest,
+    BatchGetBlocksResponse,
+    Block,
+    BlockIdentifier,
+    CommitmentState,
+    ExecuteBlockRequest,
+    GetBlockRequest,
+    GetCommitmentStateRequest,
+    UpdateCommitmentStateRequest,
 };
 use tonic::transport::Channel;
-use tracing::info;
 
+use crate::types::ExecutorCommitmentState;
+
+/// Extension trait for the ExecutionServiceClient that makes it easier to interact with the
+/// execution client.
 #[async_trait::async_trait]
-pub(crate) trait ExecutionClient {
-    async fn call_do_block(
+pub(crate) trait ExecutionClientExt {
+    async fn call_batch_get_blocks(
+        &mut self,
+        identifiers: Vec<BlockIdentifier>,
+    ) -> Result<BatchGetBlocksResponse>;
+
+    async fn call_execute_block(
         &mut self,
         prev_block_hash: Vec<u8>,
         transactions: Vec<Vec<u8>>,
-        timestamp: Option<Timestamp>,
-    ) -> Result<DoBlockResponse>;
+        timestamp: Timestamp,
+    ) -> Result<Block>;
 
-    async fn call_finalize_block(&mut self, block_hash: Vec<u8>) -> Result<()>;
+    async fn call_get_block(&mut self, identifier: BlockIdentifier) -> Result<Block>;
 
-    async fn call_init_state(&mut self) -> Result<InitStateResponse>;
-}
+    async fn call_get_commitment_state(&mut self) -> Result<ExecutorCommitmentState>;
 
-/// Represents an RpcClient. Wrapping the auto generated client here.
-pub(crate) struct ExecutionRpcClient {
-    /// The actual rpc client
-    client: ExecutionServiceClient<Channel>,
+    async fn call_update_commitment_state(
+        &mut self,
+        firm: Block,
+        soft: Block,
+    ) -> Result<ExecutorCommitmentState>;
 }
 
-impl ExecutionRpcClient {
-    /// Creates a new RPC Client
+#[async_trait::async_trait]
+impl ExecutionClientExt for ExecutionServiceClient<Channel> {
```

Comment:
where/how does the `<Channel>` generic get used? don't see it used anywhere, is it part of the autogenerated code?

Reply:
@steezeburger can you comment here, this was copied from previous PR

Reply:
yeah this is part of the proto generated stuff